### PR TITLE
Bk/7 build skeleton for mediacarousel w one video embed

### PR DESCRIPTION
**Pull request by kittbg**
#7  I completed the framework for the page.  Overall sizing is still required for the component.

### CODE REVIEW

File: server/re
Comment (ranegray) [inline server/re:1]: can you delete this file?

File: client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx [hunk 1]
```diff
@@ -1,7 +1,82 @@
...
+    fetch('http://localhost:3000/api/media/1')
+      .then(response => response.json())
+      .then(data => setData(data));
+  }, [])
+  // console.log(slides)
+
+  
+  const slides = data.map((item, index) => {
     return (
+     <div key={index} className="slide">
```
Comment (ranegray) [inline client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx:21]: using index as a key is not best practice. most of our data has id's already or you can use the url of the photo as those should be unique.

Comment (kittbg) [reply client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx]: I removed the key.

File: client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx [hunk 2]
```diff
@@ -1,7 +1,82 @@
...
+          <div className="absolute top-[50% -translate-x-0 tranlate-y-[50%] left-5 text-2xl rounded-full p-2 bg-black/20 text-white cursor-pointer">
+            <BsChevronCompactLeft onClick={nextSlide} size={30}/>
+          </div>
+
+          <div className="flex top-4 justify-center py-2">
+            {slides.map((slide, slideIndex) => (
+              <div key={slideIndex} onClick={() => goToSlide(slideIndex)} className="text-2xl cursor-pointer">
+                <RxDotFilled />
+              </div>
+            ))}
```
[the remark this answers is not in the document]
Comment (kittbg) [reply client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx]: I pushed an update.  Not sure why it did not take last night.

File: client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx [hunk 3]
```diff
@@ -13,31 +13,60 @@ function MediaCarousel() {
...
+  //     url: 'https://drive.google.com/uc?export=view&id=1hgRmHvVIuGFGKThctC9xKpPxE6UeIEVr'
+  //   }
+  // ]
+    
   const slides = data.map((item, index) => {
     return (
-     <div key={index} className="slide">
+     <div className="slide">
        <video src={item.url} autoPlay muted loop />
     </div>
```
Comment (ranegray) [inline client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx]: whenever mapping over anything you'll still need a key property. just make sure its guaranteed unique.

File: client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx [hunk 4]
```diff
@@ -58,25 +87,29 @@ return (
...
           </div>
 
-          <div className="flex top-4 justify-center py-2">
+          <div className="flex justify-center gap-2 py-4">
             {slides.map((slide, slideIndex) => (
-              <div key={slideIndex} onClick={() => goToSlide(slideIndex)} className="text-2xl cursor-pointer">
-                <RxDotFilled />
+              <div key={slideIndex} onClick={() => goToSlide(slideIndex)} className="cursor-pointer">
+                <img src="https://cdn2.unrealengine.com/egs-starwarsjedisurvivorstandardedition-respawnentertainment-g1a-07-1920x1080-320afddfd9ab.jpg?h=270&quality=medium&resize=1&w=480" alt="Jedi Survivor" className="rounded"/>
+                {/* <img src={slides.url} /> */}
```
Comment (ranegray) [inline client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx:101]: uncomment your array of urls and try using this code and you should get the correct thumbnails. 

```suggestion
          {pics.map((pic, slideIndex) => (
            <div
              key={pic.url}
              onClick={() => goToSlide(slideIndex)}
              className="cursor-pointer"
            >
              <img src={pic.url} alt="Jedi Survivor" className="rounded" />
            </div>
          ))}
```

File: client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx [hunk 5]
```diff
@@ -13,31 +13,60 @@ function MediaCarousel() {
       .then(response => response.json())
       .then(data => setData(data));
   }, [])
-  // console.log(slides)
 
-  
+  // const pics = [
```
[the remark this answers is not in the document]
Comment (ranegray) [reply client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx:17]: uncomment this for now so we can see the thumbnails.

**Review by ranegray — CHANGES_REQUESTED** [review]
Sorry but just change these few things and we should be good to press. Good job!

File: client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx (hunk 4 above)
Comment (kittbg) [reply client/src/components/ProductMain/ProductColumn/MediaCarousel/MediaCarousel.jsx]: Changes made.  Pics are resolving.

**Review by ranegray — APPROVED** [review]
All good!

**Pull request merged**